test_integration/helper/co_gateway.js

```diff
@@ -0,0 +1,444 @@
+   *
+   * @returns {Object} Object containing events and return values.
+   */
+  async confirmStakeIntent(params) {
+
+    let staker = params.staker;
+    let nonce = params.nonce;
+    let beneficiary = params.beneficiary;
+    let amount = params.amount;
+    let gasPrice = params.gasPrice;
+    let gasLimit = params.gasLimit;
+    let hashLock = params.hashLock;
+    let blockHeight = params.blockHeight;
+    let rlpParentNodes = params.rlpParentNodes;
+    let facilitator = params.facilitator;
+    let storageRoot = params.storageRoot;
+
+    await this.coGateway.setStorageRoot(
+      blockHeight,
+      storageRoot,
+      { from: this.owner },
+    );
+
+    let messageHash = await this.coGateway.confirmStakeIntent.call(
+      staker,
+      nonce,
+      beneficiary,
+      amount,
+      gasPrice,
+      gasLimit,
+      hashLock,
+      blockHeight,
+      rlpParentNodes,
+      { from: facilitator },
+    );
+
+    let tx = await this.coGateway.confirmStakeIntent(
+      staker,
+      nonce,
+      beneficiary,
+      amount,
+      gasPrice,
+      gasLimit,
+      hashLock,
+      blockHeight,
+      rlpParentNodes,
+      { from: facilitator },
+    );
+
+    let events = EventDecoder.getEvents(tx, this.coGateway);
+
+    let returnedValue = {};
+    returnedValue.messageHash_ = messageHash;
+
+    return {
+      returned_value: returnedValue,
+      events: events,
+      block_number: tx.receipt.blockNumber,
+    };
+
+  }
+
+  /**
+   * Progress mint.
+   *
+   * @param {object} params.
+   * @param {string} options.messageHash Message hash for progress mint.
+   * @param {string} options.unlockSecret Unlock secret for progress mint.
+   * @param {string} options.facilitator Facilitator address for progress mint.
+   *
+   * @returns {Object} Object containing events and return values.
+   */
+  async progressMint(params) {
+
+    let messageHash = params.messageHash;
+    let unlockSecret = params.unlockSecret;
+    let facilitator = params.facilitator;
+
+    let result = await this.coGateway.progressMint.call(
+      messageHash,
+      unlockSecret,
+      { from: facilitator },
+    );
+
+    let tx = await this.coGateway.progressMint(
+      messageHash,
+      unlockSecret,
+      { from: facilitator },
+    );
+
+    let events = EventDecoder.getEvents(tx, this.coGateway);
+
+    // Add return params in proof json data.
+    // let returnedValue = {};
+    // returnedValue.beneficiary = result.beneficiary_;
+    // returnedValue.stake_amount = result.stakeAmount_.toString(10);
+    // returnedValue.minted_amount = result.mintedAmount_.toString(10);
+    // returnedValue.reward_amount = result.rewardAmount_.toString(10);
+
+    return {
+      returned_value: result,
+      events: events,
+      block_number: tx.receipt.blockNumber,
+    };
+
+  }
+
+  /**
+   * Confirm revert stake intent.
+   *
+   * @param {object} params.
+   * @param {string} options.messageHash Message hash for confirm revert
```

Comment:
Use typedef of jsdoc for object type input params.

Reply:
Done 👍